PACKAGE = "robotics_hw1"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("test",    int_t,    0, "An Integer parameter", 0,  0, 1)
# gen.add("double_param", double_t, 1, "A double parameter",    .5, 0,   1)
# gen.add("str_param",    str_t,    2, "A string parameter",  "Hello World")
# gen.add("bool_param",   bool_t,   3, "A Boolean parameter",  True)

param_enum = gen.enum([ gen.const("Euler",      int_t, 0, "Euler integration"),
                       gen.const("Runge_Kutta",     int_t, 1, "Runge_Kutta integration")],
                     "just an enumeration"  
                     )

gen.add("E_RK", int_t, 0, "A parameter which sets the type of integration", 0, 0, 1, edit_method = param_enum)

exit(gen.generate(PACKAGE, "odom_estimator", "parameters"))